### PR TITLE
Fix bug with df.count()

### PR DESCRIPTION
**Pull request by timsaucer**
This addresses the issue in  https://github.com/rerun-io/rerun/issues/10894

Do not close that issue until test coverage is complete.

This addresses the corner case where no columns are selected, such as doing count_star with no other operations.

### CODE REVIEW

File: crates/store/re_datafusion/src/dataframe_query_common.rs [hunk 1]
```diff
@@ -406,9 +407,10 @@ pub fn align_record_batch_to_schema(
         }
     }
 
-    Ok(RecordBatch::try_new(
```
Comment (emilk) [inline crates/store/re_datafusion/src/dataframe_query_common.rs:409]: Let's ban `RecordBatch::try_new(` in `clippy.toml` to force us to fix this corner-case everywhere

**Comment by timsaucer** [comment]
@rerun-bot full-check

**Review by emilk — APPROVED** [review]
